Docs/BazelForiOSDevelopers.md

```diff
@@ -0,0 +1,522 @@
+```
+
+Bazel exposes a command line which can also do builds. In Bazel, _every_ target
+is in the global WORKSPACE and there is no notion of schemes.
+
+```
+bazel build ios-app
+```
+
+The Bazel command line has hundreds of options, which can be found in the
+[Bazel
+documentation](https://docs.bazel.build/versions/master/command-line-reference.html).
+For iOS developers, a set of useful flags is available at
+[bazel-ios-users](https://github.com/ios-bazel-users/ios-bazel-users/blob/master/UsefulFlags.md).
+
+### Configurable build attributes
+
+In addition to the default Bazel options, it's common to create custom
+configuration settings to customize builds. Together, `select` and
+`config_setting` yield configurable build attributes.
+
+For example, the following build file has conditional `copts` on `:app_store`
+
+```
+objc_library(
+   name = "some",
+   srcs = ["some.m"],
+   copts = select({
+      # For app_store we need to build `some.m` with a special copt
+      ":app_store": ["-DAPPSTORE=1"],
+      "//conditions:default": [],
+    })
+)
+config_setting(
+    name = "app_store",
+    values = { "app_store" : "true" }
+)
+```
+
+This is can be set as a define and passed to Bazel on the command line
+
+```
+bazel build ios-app --define app_store=true
+```
+
+For more information about `select` and `config_setting`, please see the [Bazel
+documentation](https://docs.bazel.build/versions/master/be/common-definitions.html#configurable-attributes).
+
+### Compiler configuration
+
+In Xcode there is a plethora of configuration options known as xcconfig that
+implicate different compiler flags.  In Bazel, `toolchains`, `objc_library`,
+`bazelrc` all configure flags. The variable `copts` in `objc_library` passes
+flags directly to the compiler. Please find canonical documentation on `copts`
+on the [Bazel
+docs](https://docs.bazel.build/versions/master/be/objective-c.html#objc_library.copts).
+Using `objc_library` to define compiler flags is useful for the per-rule level
+and many projects use macros and other layers of abstraction to [unify library
+level configuration](#Macros).
+
+
+## Extending Bazel
+
+### Starlark
+
+Bazel provides the
+[_pythonic_](https://stackoverflow.com/questions/25011078/what-does-pythonic-mean)
+programming language, Starlark. Starlark is used to implement build system
+logic and establish norms.  Generally, starlark calls into functionality that's
+implemented within Bazel. The coming segments [rules](#rules),
+[macros](#macros), and [aspects](#aspects) cover several examples.
+
+```
+.bzl file -> starlark -> bazel functions
+```
+
+Starlark enables extensibility without needing to fork the core build system.
+This boundary helps the developer to focus on their business logic instead of
+the core of the build system. For example, the Apple specific logic of bundling
+an `ios_application` is external to Bazel's core residing in the rule set,
+`rules_apple`.
```

Comment:
```suggestion
an `ios_application` is defined outside Bazel's core by `rules_apple`.
```

The first time I read this is seemed like you were implying `rules_apple` was the core of Bazel